src/box/lua/schema.lua

```diff
@@ -414,17 +419,17 @@ box.begin = function(options)
     local txn_isolation
     if options then
         timeout = options.timeout
-        txn_isolation = options.txn_isolation and
-                        normalize_txn_isolation_level(options.txn_isolation)
+        txn_isolation = options.txn_isolation
     end
     if builtin.box_txn_begin() == -1 then
         box.error()
     end
     if timeout then
         assert(builtin.box_txn_set_timeout(timeout) == 0)
     end
-    if txn_isolation and
-       builtin.box_txn_set_isolation(txn_isolation) ~= 0 then
+    txn_isolation = normalize_txn_isolation_level(txn_isolation)
+    assert(txn_isolation ~= nil)
+    if builtin.box_txn_set_isolation(txn_isolation) ~= 0 then
```

Comment:
Do you really need to patch `box.begin()`? I think it should do just fine without any modifications.

Reply:
Yes, I do, because now default isolation can be "linearizable". And this level requires extra actions (wait_linearization_point) once it is set.
Previously `txn->isolation` was set to default value in `txn_begin`, so when `box.begin()` had no isolation, it just left the `txn->isolation` value intact. But this can't be done now when default isolation is 'linearizable'

Reply:
>I think it should do just fine without any modifications.

Agreed

Reply:
I've reverted all the extra changes:
```diff
diff --git a/src/box/lua/schema.lua b/src/box/lua/schema.lua
index f5f045a68..2eead2d3a 100644
--- a/src/box/lua/schema.lua
+++ b/src/box/lua/schema.lua
@@ -388,9 +388,6 @@ box.internal.txn_isolation_level_map = txn_isolation_level_map
 
 -- Convert to numeric the value of txn isolation level, raise if failed.
 local function normalize_txn_isolation_level(txn_isolation)
-    if txn_isolation == nil then
-        return 0
-    end
     txn_isolation = txn_isolation_level_map[txn_isolation]
     if txn_isolation == nil then
         box.error(box.error.ILLEGAL_PARAMS,
@@ -419,7 +416,8 @@ box.begin = function(options)
     local txn_isolation
     if options then
         timeout = options.timeout
-        txn_isolation = options.txn_isolation
+        txn_isolation = options.txn_isolation and
+                        normalize_txn_isolation_level(options.txn_isolation)
     end
     if builtin.box_txn_begin() == -1 then
         box.error()
@@ -427,9 +425,8 @@ box.begin = function(options)
     if timeout then
         assert(builtin.box_txn_set_timeout(timeout) == 0)
     end
-    txn_isolation = normalize_txn_isolation_level(txn_isolation)
-    assert(txn_isolation ~= nil)
-    if builtin.box_txn_set_isolation(txn_isolation) ~= 0 then
+    if txn_isolation and
+       builtin.box_txn_set_isolation(txn_isolation) ~= 0 then
         box.rollback()
         box.error()
     end

```